src/rgw/rgw_cr_rados.cc

```diff
@@ -713,20 +712,22 @@ int RGWAsyncFetchRemoteObj::_send_request(const DoutPrefixProvider *dpp)
           }
         }
         
+        // NOTE: we create a mutable copy of bucket.get_tenant as the get_notification function expects a std::string&, not const
```

Comment:
i think you can just change the function to use `const std::string&`
nothing should change the tenant inside the function

Reply:
Changing the function to use `const std::string&` has a lot of other side effects and changes to other code, namely the class `reservation_t` from `rgw_notify.h` will need to be updated to have a `const` tenant, as well as any class that deals with notifications (which is a couple)
Ideally I think it should be a separate PR if anything as it touches a lot of unrelated files, and it isn't necessarily required for this one

Reply:
agree. making all of them const is a good thing.
but lets keep it out of scope of this PR